### DISCUSSION TITLE
problems adding support for Nucleo-L4P5

### DISCUSSION THREAD

**Discussion by PerennialNovice** [stm32duino/Arduino_Core_STM32#1521] [Q&A] [answered]
Hi,
I try to add support for Nucleo-L4P5 and started off adding a generic variant following all the steps described in the [wiki article](https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-%28board%29).
I finally tried to execute the [CheckVariant sketch](https://github.com/stm32duino/STM32Examples/tree/main/examples/NonReg/CheckVariant) sketch but could not get serial output.
To rule out any of my setup other than the board itself, I tried the CheckVariant sketch on Nucleo-L4R5 which (after some trouble #1517) worked out fine (using SWD download for some reason unknown to me needs a manual reset).
When I select Generic-L4R5 however, no serial output was generated.
I then copied the clock setup from Nucleo to the generic variant but this did not change anything (the new clock setup gets used - I check by inserting a `#pragma message` to the function).

Seems that there is more to the Nucleo-implemantation that needs to be done to have the generic variant work...

What am I missing?
Very happy for every hint and answer!

**Comment by fpistm** [comment, marked as answer]
>  When I select Generic-L4R5 however, no serial output was generated.

If you use the generic with the Nucleo then you have to redefine the default Serial pin.
As default Serial for Nucleo L4R5 are:
https://github.com/stm32duino/Arduino_Core_STM32/blob/6479b8a0936852c052c16f79606cd1192fc9dcc1/variants/STM32L4xx/L4R5Z(G-I)T_L4R7ZIT_L4S5ZIT_L4S7ZIT/variant_NUCLEO_L4R5ZI.h#L218-L229
While generic define them as:
https://github.com/stm32duino/Arduino_Core_STM32/blob/6479b8a0936852c052c16f79606cd1192fc9dcc1/variants/STM32L4xx/L4R5Z(G-I)T_L4R7ZIT_L4S5ZIT_L4S7ZIT/variant_generic.h#L229-L240

To change them you can use the `setRx()` and `setTx()` API before the `begin()` or connect a FTDI to the correct pins.

**Reply by PerennialNovice** [reply]
Thank you very much!

I feel a bit ashamed not having thought of the pins myself...
